kv/kvserver/raftlog: TestIterator failed

kv/kvserver/raftlog.TestIterator [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/17684533?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/17684533?buildTab=artifacts#/) on release-23.1 @ [d8903402a434ef4bc4de2f55201a1b9957c61ce4](https://github.com/cockroachdb/cockroach/commits/d8903402a434ef4bc4de2f55201a1b9957c61ce4):

Fatal error:

```
panic: logging already active; first use:
```
Stack: 

```
goroutine 898 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/log.setActive()
	github.com/cockroachdb/cockroach/pkg/util/log/clog.go:413 +0xa7
github.com/cockroachdb/cockroach/pkg/util/log.(*loggerT).outputLogEntry(0xc000285680, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}}, 0x1806a33150d576e2, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/clog.go:245 +0x50
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x1d8fcc0, 0xc00014d9b0}, 0x4, 0x1, 0x3, 0x78?, {0x19ffaf5, 0x2}, {0xc000637450, 0x1, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:101 +0x605
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:34
github.com/cockroachdb/cockroach/pkg/util/log.loggerStorage.InfofDepth(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:2044
github.com/cockroachdb/cockroach/pkg/storage.pebbleLogger.Infof({{0x1d8fcc0?, 0xc00014d9b0?}, 0x0?}, {0x19ffaf5?, 0x41e527?}, {0xc000637450?, 0x1752880?, 0x1?})
	github.com/cockroachdb/cockroach/pkg/storage/pebble_logger_and_tracer.go:23 +0x56
github.com/cockroachdb/pebble.MakeLoggingEventListener.func13({0x6?})
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:634 +0x9e
github.com/cockroachdb/pebble.TeeEventListener.func13({0xc0000fb400?})
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:709 +0x42
github.com/cockroachdb/pebble.(*DB).collectTableStats(0xc0000fb400)
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:121 +0x28e
created by github.com/cockroachdb/pebble.(*DB).maybeCollectTableStatsLocked
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:46 +0x65
 [recovered]
	panic: logging already active; first use:
goroutine 898 [running]:
runtime/debug.Stack()
	GOROOT/src/runtime/debug/stack.go:24 +0x65
github.com/cockroachdb/cockroach/pkg/util/log.setActive()
	github.com/cockroachdb/cockroach/pkg/util/log/clog.go:413 +0xa7
github.com/cockroachdb/cockroach/pkg/util/log.(*loggerT).outputLogEntry(0xc000285680, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}}, 0x1806a33150d576e2, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/clog.go:245 +0x50
github.com/cockroachdb/cockroach/pkg/util/log.logfDepthInternal({0x1d8fcc0, 0xc00014d9b0}, 0x4, 0x1, 0x3, 0x78?, {0x19ffaf5, 0x2}, {0xc000637450, 0x1, ...})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:101 +0x605
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth(...)
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:34
github.com/cockroachdb/cockroach/pkg/util/log.loggerStorage.InfofDepth(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-fastbuild/bin/pkg/util/log/log_channels_generated.go:2044
github.com/cockroachdb/cockroach/pkg/storage.pebbleLogger.Infof({{0x1d8fcc0?, 0xc00014d9b0?}, 0x0?}, {0x19ffaf5?, 0x41e527?}, {0xc000637450?, 0x1752880?, 0x1?})
	github.com/cockroachdb/cockroach/pkg/storage/pebble_logger_and_tracer.go:23 +0x56
github.com/cockroachdb/pebble.MakeLoggingEventListener.func13({0x6?})
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:634 +0x9e
github.com/cockroachdb/pebble.TeeEventListener.func13({0xc0000fb400?})
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/event.go:709 +0x42
github.com/cockroachdb/pebble.(*DB).collectTableStats(0xc0000fb400)
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:121 +0x28e
created by github.com/cockroachdb/pebble.(*DB).maybeCollectTableStatsLocked
	github.com/cockroachdb/pebble/external/com_github_cockroachdb_pebble/table_stats.go:46 +0x65
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestIterator
I241110 14:55:38.975426 898 3@pebble/event.go:709  [s?,pebble] 1  [JOB 4] all initial table stats loaded
--- FAIL: TestIterator (0.00s)
```
</p>
</details>

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestIterator.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-44254

--- a/pkg/kv/kvserver/raftlog/encoding_test.go
+++ b/pkg/kv/kvserver/raftlog/encoding_test.go
@@ -12,6 +12,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/kvflowcontrolpb"
 	"github.com/cockroachdb/cockroach/pkg/util/admission/admissionpb"
 	"github.com/cockroachdb/cockroach/pkg/util/humanizeutil"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 	"github.com/stretchr/testify/require"
@@ -31,6 +32,7 @@ import (
 //	RaftAdmissionMetaOverhead/bytes=1.0_MiB,raft-ac-10   148µs ± 4%   151µs ± 5%     ~     (p=0.095 n=5+5)
 //	RaftAdmissionMetaOverhead/bytes=2.0_MiB,raft-ac-10   290µs ± 3%   292µs ± 1%     ~     (p=0.151 n=5+5)
 func BenchmarkRaftAdmissionMetaOverhead(b *testing.B) {
+	defer leaktest.AfterTest(b)()
 	defer log.Scope(b).Close(b)
 
 	const KiB = 1 << 10

--- a/pkg/kv/kvserver/raftlog/entry_bench_test.go
+++ b/pkg/kv/kvserver/raftlog/entry_bench_test.go
@@ -9,9 +9,12 @@ package raftlog
 import (
 	"fmt"
 	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 )
 
 func BenchmarkNewEntry(b *testing.B) {
+	defer leaktest.AfterTest(b)()
 	ent, metaB := mkBenchEnt(b)
 	b.ResetTimer()
 	for _, fromRawValue := range []bool{false, true} {

--- a/pkg/kv/kvserver/raftlog/entry_test.go
+++ b/pkg/kv/kvserver/raftlog/entry_test.go
@@ -9,11 +9,14 @@ package raftlog
 import (
 	"testing"
 
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/stretchr/testify/require"
 	"go.etcd.io/raft/v3/raftpb"
 )
 
 func TestNewEntry(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
 	// TODO(replication): Add more cases.
 	testcases := map[string]struct {
 		data        []byte

--- a/pkg/kv/kvserver/raftlog/iter_test.go
+++ b/pkg/kv/kvserver/raftlog/iter_test.go
@@ -16,6 +16,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverpb"
 	"github.com/cockroachdb/cockroach/pkg/storage"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 	"github.com/stretchr/testify/require"
@@ -144,9 +145,11 @@ func (it *modelIter) Entry() raftpb.Entry {
 }
 
 func TestIteratorEmptyLog(t *testing.T) {
+	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
 	eng := storage.NewDefaultInMemForTesting()
+	defer eng.Close()
 	for _, hi := range []uint64{0, 1} {
 		it := NewIterator(rangeID, eng, IterOptions{Hi: hi})
 		ok, err := it.SeekGE(0)
@@ -159,6 +162,7 @@ func TestIteratorEmptyLog(t *testing.T) {
 // TestIterator sets up a few raft logs and iterates all conceivable chunks of
 // them with both the real and a model iterator, comparing the results.
 func TestIterator(t *testing.T) {
+	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
 	for _, tc := range []struct {

